### DISCUSSION TITLE
Schema Location

### DISCUSSION THREAD

**Discussion by ChristianSauer** [wixtoolset/issues#7182] [Questions]
Is there a public schema location for the various schemas?
E.g. http://wixtoolset.org/schemas/v4/wxs/util does not exist but I would love to have the real XSD so that I can teach Rider how the valid XML must look like.

**Comment by robmen** [comment]
The XSDs are in the web project. They are used to generate documentation.

**Reply by domischell** [reply]
Hi,

I would also be interested in the schema... the url gets forwarded to a song from The Smashing Pumpkins (rickrolled? 😉) which is very confusing.

It would be great to have the schema for Intellisense in Visual Studio, so maybe someone could upload it?
[Update: I noticed that Intellisense is working by some black magic. I'm very confused now, what is the purpose of this xmlns element in the wxs file?]

Thanks and keep up the great work! I love where v4 is heading.
Dom

**Reply by robmen** [reply]
I thought about using Rick Astley but "Bullet with Butterfly Wings" acutely captures the dichotomy between OSS maintainers and consumers. Billy Corgan was writing about his place in the record industry, but he could have just as well been the maintainer of a successful mid-sized Open Source project. 

> It would be great to have the schema for Intellisense in Visual Studio, so maybe someone could upload it?
> [Update: I noticed that Intellisense is working by some black magic. I'm very confused now, what is the purpose of this xmlns element in the wxs file?]

The xmlns _attribute_ is an opaque unique identifier used to connect an XML document to its schema. URIs are commonly used in xmlns attributes because the Internet registrars ensure domain names are globally unique. Intellisense works in Visual Studio because [HeatWave](https://www.firegiant.com/wix/heatwave/) installs the XSDs into VS so it can "connect" the XSD to your XML document via the xmlns attribute value. It has nothing to do with the Internet necessarily.

> I would also be interested in the schema.

As I said above, the XSD files are in the web project.
